packages/app/components/stackbox/FrequencyOptionsCard.tsx

```diff
@@ -0,0 +1,132 @@
+"use client";
+
+import { ChangeEvent, useEffect, useState } from "react";
+
+import { add } from "date-fns";
+import { twMerge } from "tailwind-merge";
+
+import { FREQUENCY_OPTIONS } from "@/models";
+import { BodyText, RadioButton, TextInput } from "@/ui";
+
+interface FrequencyOptionsCardProps {
+  frequency: FREQUENCY_OPTIONS;
+  setEndDate: (date: Date) => void;
+}
+
+const defaultFrequencyOptions = {
+  [FREQUENCY_OPTIONS.hour]: ["6", "12", "24"],
+  [FREQUENCY_OPTIONS.day]: ["7", "15", "30"],
+  [FREQUENCY_OPTIONS.week]: ["4", "8", "12"],
+  [FREQUENCY_OPTIONS.month]: ["2", "6", "12"],
+};
+const maxCustomFrequencies = {
+  [FREQUENCY_OPTIONS.hour]: 96,
+  [FREQUENCY_OPTIONS.day]: 365,
+  [FREQUENCY_OPTIONS.week]: 52,
+  [FREQUENCY_OPTIONS.month]: 24,
+};
+const postiveIntegerOnly = /^[1-9][0-9]*$/;
```

Comment:
Maybe we could add Regex, so it's super clear

```suggestion
const postiveIntegerOnlyRegex = /^[1-9][0-9]*$/;
```